### ISSUE TITLE
Auto3DSeg ensembler module doesn't work with Bundle Algorithms generated by the HPO Module

### ISSUE THREAD

**Issue by mingxin-zheng** [Project-MONAI/MONAI#5558]
**Describe the bug**

Auto3DSeg HPO module generates bundle algos and append the hyperparameters in the name of the bundle root folder

For examples, to perform a grid_search on learning rate (lr), folders will be like
. 
├── dints_0_lr_0.0001
├── dints_0_lr_0.001
├── dints_0_lr_0.01
└── dints_0_lr_0.1

In the integration test and some possible usage scenarios, these bundle algorithms need to be ensembled by fold, and the HPO folder name would break Auto3DSeg AutoRunner.

Related line: https://github.com/Project-MONAI/MONAI/blob/72130602340847e15f2042b2bb1b64f6f46eeb93/monai/apps/auto3dseg/ensemble_builder.py#L234

Related issue: #5523 #5555
